blockchain/validators/src/error.rs

```diff
@@ -0,0 +1,143 @@
+use displaydoc::Display;
+use hex::FromHexError;
+use mc_attest_core::VerifyError;
+use mc_attest_verifier::Error as VerifierError;
+use mc_blockchain_types::BlockIndex;
+use mc_common::ResponderId;
+use mc_crypto_keys::KeyError;
+use pem::PemError;
+use serde_json::Error as JsonError;
+use std::{io::Error as IoError, path::PathBuf};
+use toml::de::Error as FromTomlError;
+
+#[derive(Debug, Display)]
+pub enum ValidationError {
+    /// Failed to parse key: {0}
+    Key(KeyError),
+
+    /// Unrecognized public key.
+    UnknownPubKey,
+
+    /// Expired/Invalid public key.
+    InvalidPubKey,
+
+    /// Signature error: {0}
+    Signature(String),
+
+    /// No valid AVR found for signing key {0}
+    AvrNotFound(String),
+
+    /// Block Signer key: {0} does not match public key from AVR: {0}
+    AvrSignerKeyMismatch(String, String),
+
+    /// Signer key contained in AVR was invalid: {0},
+    AvrKeyData(VerifyError),
+
+    /// AVR failed to verify against intel roots of trust {0},
+    InvalidAvr(VerifierError),
+
+    /// Signing Key is out of range for block {0}
+    BlockSigningKeyExpired(BlockIndex),
+
+    /// ResponderId: {0} does not match expected ResponderId: {1}
+    ResponderMismatch(ResponderId, ResponderId),
+
+    /// No Block Signature found at index: {0},
+    NoBlockSignature(BlockIndex),
+
+    /// Other error: {0}
+    Other(String),
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum ParseError {
+    /// Unrecognized extension in '{0}'
+    UnrecognizedExtension(PathBuf),
+
+    /// Invalid pub_key value: {0}
+    InvalidPubKeyValue(String),
+
+    /// Failed to parse key: {0}
+    Key(KeyError),
+
+    /// I/O error: {0}
+    Io(String),
+
+    /// Failed to parse hexadecimal string: {0}
+    Hex(FromHexError),
+
+    /// Failed to parse PEM: {0}
+    Pem(PemError),
+
+    /// Invalid PEM tag: {0}
+    InvalidPemTag(String),
+
+    /// Failed to parse TOML: {0}
+    Toml(String),
+
+    /// Failed to parse JSON: {0}
+    Json(String),
+
+    /// AVR failed to verify against intel roots of trust {0}
+    InvalidAvr(VerifierError),
+
+    /// Signer key contained in AVR was invalid: {0}
+    AvrKeyData(VerifyError),
```

Comment:
Verifier errors should be validation errors, not parse errors

Reply:
Changed up the errors a bit